resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/builders/InterruptingBuilderTest.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Patrick Ziegler and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Patrick Ziegler - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.core.tests.internal.builders;
+
+import org.eclipse.core.internal.resources.Workspace;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+
+/**
+ * Tests that the AutoBuild job doesn't raise its interrupt flag when doing
+ * workspace modifications.
+ */
+public class InterruptingBuilderTest extends AbstractBuilderTest {
+
+	private static final String TEST_PROJECT_NAME = "WorkerProject";
+	private static final String TEST_FILE_NAME = "test.txt";
+	private IProject project;
+	private IFile file;
+
+	public InterruptingBuilderTest() {
+		super("Interrupting Builder");
+	}
+
+	@Override
+	public void setUp() throws Exception {
+		super.setUp();
+		setAutoBuilding(false);
```

Comment:
Typically you have to wait till autobuild job finished in junit tests after disabling it